src/main.rs

```diff
@@ -458,41 +469,64 @@ async fn enter_redraw_loop(
 	}
 }
 
+const MIN_DEBOUNCE_DELAY: Duration = Duration::from_millis(50);
+
 fn on_notify_ev(
 	to_tui_tx: flume::Sender<Result<RenderInfo, RenderError>>,
 	to_render_tx: flume::Sender<RenderNotif>,
-	file_name: OsString
+	file_name: OsString,
+	debounce_delay: Duration
 ) -> impl Fn(notify::Result<Event>) {
-	move |res| match res {
-		// If we get an error here, and then an error sending, everything's going wrong. Just give
-		// up lol.
-		Err(e) => to_tui_tx.send(Err(RenderError::Notify(e))).unwrap(),
-		// TODO: Should we match EventKind::Rename and propogate that so that the other parts of the
-		// process know that too? Or should that be
-		Ok(ev) => {
-			// We only watch the parent directory (see the comment above `watcher.watch` in `fn
-			// main`) so we need to filter out events to only ones that pertain to the single file
-			// we care about
-			if !ev
-				.paths
-				.iter()
-				.any(|path| path.file_name().is_some_and(|f| f == file_name))
-			{
-				return;
+	let debounce_delay = debounce_delay.max(MIN_DEBOUNCE_DELAY);
+	let last_event: Mutex<Result<(), RenderError>> = Mutex::new(Ok(()));
+	let last_event = Arc::new(last_event);
+
+	let debouncer = EventDebouncer::new(debounce_delay, {
+		let last_event = last_event.clone();
+		move |()| {
+			let event = mem::replace(&mut *last_event.lock().unwrap(), Ok(()));
+			match event {
+				Ok(()) => to_render_tx.send(RenderNotif::Reload).unwrap(),
+				Err(e) => to_tui_tx.send(Err(e)).unwrap()
 			}
+		}
+	});
+
+	move |res| {
+		let event = match res {
+			// If we get an error here, and then an error sending, everything's going wrong. Just give
+			// up lol.
```

Comment:
I think this comment needs to be pulled up to inside the closure in `EventDebouncer::new` since it pertains to the sending of the message, not the creation of it.